src/transport_picoquic.cpp

```diff
@@ -22,908 +25,884 @@
+    if (cbNotifyThread.joinable()) {
+        logger.log(LogLevel::info, "Closing transport callback notifier thread");
+        cbNotifyThread.join();
+    }
 
-  logger.log(LogLevel::info, "done closing transport threads");
+    _timer.reset();
+    logger.log(LogLevel::info, "done closing transport threads");
 
-  picoquic_config_clear(&config);
+    picoquic_config_clear(&config);
 }
 
-
 TransportStatus
-PicoQuicTransport::status() const {
-  return transportStatus;
+PicoQuicTransport::status() const
+{
+    return transportStatus;
 }
 
-void PicoQuicTransport::setStatus(TransportStatus status) {
-  transportStatus = status;
+void
+PicoQuicTransport::setStatus(TransportStatus status)
+{
+    transportStatus = status;
 }
 
-StreamId PicoQuicTransport::createStream(const TransportContextId &context_id, bool use_reliable_transport) {
+StreamId
+PicoQuicTransport::createStream(const TransportContextId& context_id, bool use_reliable_transport)
+{
+    logger.log(LogLevel::debug, "App requests to create new stream");
 
-  const auto &iter = active_streams.find(context_id);
-  if (iter == active_streams.end()) {
-    throw std::invalid_argument("Invalid context id, cannot create stream");
-  }
+    const auto& iter = active_streams.find(context_id);
+    if (iter == active_streams.end()) {
+        throw std::invalid_argument("Invalid context id, cannot create stream");
+    }
 
-  const auto &cnx_stream_iter = iter->second.find(0);
-  if (cnx_stream_iter == iter->second.end()) {
-    throw std::invalid_argument("Missing primary connection zero stream, cannot create streams");
-  }
+    const auto& cnx_stream_iter = iter->second.find(0);
+    if (cnx_stream_iter == iter->second.end()) {
+        throw std::invalid_argument("Missing primary connection zero stream, cannot create streams");
+    }
 
-  if (!use_reliable_transport)
-    return ::make_datagram_stream_id(_is_server_mode, _is_bidirectional);
+    if (!use_reliable_transport)
+        return ::make_datagram_stream_id(_is_server_mode, _is_bidirectional);
 
-  next_stream_id = ::make_stream_id(next_stream_id + 4, _is_server_mode, _is_bidirectional);
+    next_stream_id = ::make_stream_id(next_stream_id + 4, _is_server_mode, _is_bidirectional);
 
-  PicoQuicTransport::StreamContext *stream_cnx = createStreamContext(
-          cnx_stream_iter->second.cnx,
-          next_stream_id);
+    PicoQuicTransport::StreamContext *stream_cnx = createStreamContext(
+      cnx_stream_iter->second.cnx,
+      next_stream_id);
 
-  TransportContextId tcid = context_id;
-  StreamId stream_id = next_stream_id;
+    TransportContextId tcid = context_id;
+    StreamId stream_id = next_stream_id;
 
-  cbNotifyQueue.push([&]() {
-    delegate.on_new_stream(tcid, stream_id);
-  });
+    cbNotifyQueue.push([&]() { delegate.on_new_stream(tcid, stream_id); });
 
-  return stream_cnx->stream_id;
+    // Set next stream ID
+    next_stream_id += 4; // Increment by 4, stream type is first 2 bits
```

Comment:
Looks like there was a blip when merging the stream changes, this increment is being handled on line 549. Not a big deal, but this is incrementing twice.